sdk/ledger-service/http-json/src/main/scala/com/digitalasset/http/PackageService.scala

```diff
@@ -375,28 +375,53 @@ object PackageService {
   final case class ContractTypeIdMap[CtId[T] <: ContractTypeId[T]](
       all: Map[RequiredPkg[CtId], ResolvedOf[CtId]],
       unique: Map[NoPkg[CtId], ResolvedOf[CtId]],
-      nameIds: Map[Ref.PackageName, String], // TODO(ETX-451): Multiple package ids per package name
+      nameIds: Map[Ref.PackageName, NonEmpty[Seq[String]]],
+      idNames: PackageNameMap,
   ) {
-    private def packageRefToPackageId(pkgRef: String): Option[String] = pkgRef match {
-      case s"#${name}" => nameIds.get(asPackageName(name))
-      case pkgId => Some(pkgId)
+    private[http] def toContractTypeRef[R <: ContractTypeId.Resolved](
+        id: R
+    ): Option[ContractTypeRef[R]] = {
+      id.packageId match {
+        case s"#${name}" =>
+          for {
+            pkgIdsForName <- nameIds.get(asPackageName(name))
+            pkgIdsForCtId <- NonEmpty.from(
+              pkgIdsForName.filter(pkgId =>
+                all.contains(id.copy(packageId = pkgId).asInstanceOf[RequiredPkg[CtId]])
+              )
+            )
+            (kpn, _) <- idNames.get(id.packageId)
+          } yield ContractTypeRef(id, pkgIdsForCtId, kpn)
+        case pkgId =>
+          Some(ContractTypeRef(id, NonEmpty(Seq, pkgId), KeyPackageName.empty))
```

Comment:
This makes me nervous from a modelling PoV if we are assuming this version of http-json will only work with package-names then there should never be a need to use NoPackageName (empty).